Refactor experiment result class

douglasjacobsen opened the pull request:
This merge splits the init method of the experiment result class into an init (which doesn't do anything beyond creating the attributes of the result class), and a finalize method (which correctly sets the attributes). This gets around issues where experiments (namely, repeats under parent experiments) were not setup / run and are considered invalid because they didn't have a valid results object.

rfbgo reviewed (dismissed):
Thanks for this updated change. This is a nice fix

douglasjacobsen committed:
Refactor initialization of experiment results

This commit refactors the initialization of experiment results.
Previously, the init happened really close to when the results were
populated. This caused some experiments (i.e. children of repeats that
were not executed) to have invalid results objects that were still
accessed (i.e. by their parents). This caused an exception to be thrown
in this (arguably rare) instance.

This refactor avoids this by splitting experiment results into an init
and finalize step.
